src/js/api/shipment.js

```diff
@@ -0,0 +1,319 @@
+ * - 'skip' - The number of the first record to retrieve from the results.
+ * - 'limit' - The number of results to retrieve.
+ * @returns {Promise} The shipments result list.
+ */
+ripe.Ripe.prototype.getShipmentsP = function(options) {
+    return new Promise((resolve, reject) => {
+        this.getShipments(options, (result, isValid, request) => {
+            isValid ? resolve(result) : reject(new ripe.RemoteError(request, null, result));
+        });
+    });
+};
+
+/**
+ * Gets a shipment by number.
+ *
+ * @param {Number} number The number of the shipment to find by.
+ * @param {Object} options An object of options to configure the request.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.getShipment = function(number, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const url = `${this.url}shipments/${number}`;
+    options = Object.assign(options, {
+        url: url,
+        method: "GET",
+        auth: true
+    });
+    options = this._build(options);
+    return this._cacheURL(options.url, options, callback);
+};
+
+/**
+ * Gets an shipment by number.
+ *
+ * @param {Number} number The number of the shipment to find by.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {Promise} The shipment requested by number.
+ */
+ripe.Ripe.prototype.getShipmentP = function(number, options) {
+    return new Promise((resolve, reject) => {
+        this.getShipment(number, options, (result, isValid, request) => {
+            isValid ? resolve(result) : reject(new ripe.RemoteError(request, null, result));
+        });
+    });
+};
+
+/**
+ * Creates a shipment on RIPE Core.
+ *
+ * @param {Object} shipment The shipment object, with options such as:
+ *  - 'status' - The shipment status.
+ *  - 'description' - A brief description of this shipment.
+ *  - 'weight' - Weight of the package being shipped.
+ *  - 'weight_units' - Weight units (e.g. "kilograms" or "pounds").
+ *  - 'courier' - The courier for this shipment.
+ *  - 'service' - The transportation service used.
+ *  - 'package' - The package used.
+ *  - 'tracking_number' - The tracking number associated with this shipment.
+ *  - 'tracking_url' - The tracking URL associated with this shipment.
+ *  - 'carbon_neutral' - If this shipment should be carbon neutral.
+ *  - 'access_point_delivery' - If this shipment should be sent to an access point (e.g. "no", "optional" or "mandatory").
+ *  - 'pickup' - Pickup information.
+ *    - 'offset' - Time offset for pickup scheduling relative to shipment creation (millis).
+ *    - 'ready_time' - Opening hours for shipper facility (millis).
+ *    - 'close_time' - Closing hours for shipper facility (millis).

```

Comment:
We usually write the whole word `milliseconds` instead of `millis`. Better to follow the presumed standard we have regarding this